lib/logger/lib/logger/backends/console.ex

```diff
@@ -97,4 +163,52 @@ defmodule Logger.Backends.Console do
   defp color_event(data, level, %{enabled: true} = colors) do
     [IO.ANSI.format_fragment(Map.fetch!(colors, level), true), data | IO.ANSI.reset]
   end
+
+  defp log_buffer(%{buffer_size: 0, buffer: []} = state), do: state
+
+  defp log_buffer(state) do
+    %{device: device, buffer: buffer, buffer_ts: buffer_ts} = state
+    %{state | ref: async_io(device, buffer), buffer: [], buffer_size: 0,
+      buffer_ts: nil, last_ts: buffer_ts}
```

Comment:
I am also fine with calling `Logger.Utils.timestamp` when we need to log our own messages if it means we don't need to do the `ts` dance.